### PR TITLE
Added solution for Edit Distance

### LINKED ISSUE #439: Edit Distance in DP
## 💥 Proposal

Find minimum number of edits ( Insert, Remove, Replace) required to convert one string to another.

Suppose we have string1 and string2

Input:   string1 = "adft", string2 = "adsft"
Output:  1
We can convert string1 into string2 by inserting a 's'.

Input:   string1 = "cat", string2 = "car"
Output:  1
We can convert string1 into string2 by replacing 't' with 'r'.

Input:   string1 = "sundays", string2 = "monday"
Output:  3

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?
Yes

**Language Supported:**
- C [@nutanaarohi123]
- CPP [@Joshitha18]
- Python [@santushtisharma10]
- C#
- Java [@AltCtrlDel1999]
- GO
- JavaScript

### PR DESCRIPTION
**Pull request by Joshitha18**
Fixes:  #439
Edit Distance

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Added DP solution for Edit Distance Problem

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#439

### CODE REVIEW

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @Joshitha18 Good work! The code is working fine! Please add one input and output sample below the code section in commented form and it will be good to go.

Also please provide a bit of description about the problem your PR is going to solve or simpley link the related issue using `Fixes: #[number of the issue without these braces]` so that we can know what the PR is actually for :)

**Comment by Joshitha18** [comment]
@Kajol-Kumari Added requested changes

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :)

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]
print a message before input describing the kind of input user have to enter 
Also, print the output with a message

**Comment by Joshitha18** [comment]
@plazzy99 Added requested changes

**Review by vatsalkesarwani12 — APPROVED** [review]
Request changes done 💯

**Comment by vatsalkesarwani12** [comment]
@Joshitha18 also try to make a change to the original file and don't delete and add another file.

Follow few online tutorials to learn Github

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]
I would like two things here,

1 - Implement the string algorithms (This is DP-String Manipulation) using the String Class in C++ because In most of the interviews they ask to use string class because that is tricky and something not present in procedural languages like C unlike char array which is common in C and C++.
2 - Add the recursive approach as well. Although it's a DP problem, but the recursive approach helps the learner in deriving to the DP solution.

Your code is good, but we should focus on writing better code using some better functionalities in CPP like string class and OOP.

**Comment by Joshitha18** [comment]
> I would like two things here,
> 
> 1 - Implement the string algorithms (This is DP-String Manipulation) using the String Class in C++ because In most of the interviews they ask to use string class because that is tricky and something not present in procedural languages like C unlike char array which is common in C and C++.
> 2 - Add the recursive approach as well. Although it's a DP problem, but the recursive approach helps the learner in deriving to the DP solution.
> 
> Your code is good, but we should focus on writing better code using some better functionalities in CPP like string class and OOP.

@Abhijit2505 I'll add recursive approach but I don't understand what u mean by 'Implement string Algorithm' for this above code. Can u be more elaborate or give some examples?

**Comment by Joshitha18** [comment]
@Abhijit2505 Added changes

**Comment by Joshitha18** [comment]
@Abhijit2505 ?

**Comment by Joshitha18** [comment]
@Abhijit2505 Please approve this. I made requested changes

File: C-Plus-Plus/dp/edit_distance.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+//Edit distance DP approach
+int edit_distance_dp (string &x, string &y)
+{
+    int m=x.length();
+    int n=y.length();
+
+    int dp[101][101];
+
```
Comment (Abhijit2505) [inline C-Plus-Plus/dp/edit_distance.cpp:27]: Please change it into dynamic memory allocation, instead of some fixed size.

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]

**Comment by Joshitha18** [comment]
@Abhijit2505 done

**Pull request merged**